Mathlib/NumberTheory/Cyclotomic/Rat.lean

```diff
@@ -274,6 +275,68 @@ theorem subOneIntegralPowerBasis'_gen [hcycl : IsCyclotomicExtension {p} ℚ K]
   @subOneIntegralPowerBasis_gen p 1 K _ _ _ _ (by convert hcycl; rw [pow_one]) (by rwa [pow_one])
 #align is_primitive_root.sub_one_integral_power_basis'_gen IsPrimitiveRoot.subOneIntegralPowerBasis'_gen
 
+/-- `ζ - 1` is prime if `p ≠ 2` and `ζ` is a primitive `p ^ (k + 1)`-th root of unity.
+  See `zeta_sub_one_prime` for a general statement. -/
+theorem zeta_sub_one_prime_ne_two [IsCyclotomicExtension {p ^ (k + 1)} ℚ K]
+    (hζ : IsPrimitiveRoot ζ ↑(p ^ (k + 1))) (hodd : p ≠ 2) :
+    Prime (⟨ζ - 1, Subalgebra.sub_mem _ (hζ.isIntegral (p ^ _).pos)
```

Comment:
Why not make a definition/abbrev for ζ as an element in the subalgebra (of algebraic integers?) and use subtraction in the subalgebra?

Reply:
In most application `ζ` will be an element of `K` (given by `zeta`). Indeed something I can add right now is that the generator of [IsPrimitiveRoot.subOneIntegralPowerBasis](https://leanprover-community.github.io/mathlib4_docs/Mathlib/NumberTheory/Cyclotomic/Rat.html#IsPrimitiveRoot.subOneIntegralPowerBasis) is prime.

Reply:
Can you make an def/abbrev `IsPrimitiveRoot.toIntegral : 𝓞 K := ⟨ζ , hζ.isIntegral (p ^ _).pos⟩` so you can write `hζ.toIntegral - 1` which should be defeq to the long expression
```
⟨ζ - 1, Subalgebra.sub_mem _ (hζ.isIntegral (p ^ _).pos) (Subalgebra.one_mem _)⟩ : 𝓞 K
```

Reply:
Ah yes, using an abbreviation seems a good idea.